backend/server.py

```diff
@@ -321,63 +355,72 @@ def add_inference():
     task_id = edge_node.request_inference(inference)
 
     print(f'Request sent with task id: {task_id}. Waiting inference...')
-    # Wait to received solution
-    waiter_inference.wait()
-    print('Inference received.')
 
-    return jsonify(inference_data)
+    # Wait to received solution
+    reason=waiter_inference.wait(timeout)
+    if reason==AwakeReason.timeout:
+        print('Timeout reached.')
+        return jsonify({'Error': 'Timeout reached.'})
+    else:
+        print('Inference received.')
+        return jsonify(inference_data)
 
 @app.route('/context_broker/fiware', methods=['GET', 'POST'])
 def create_fiware_node():
-
     global fiware_node_content
     fiware_node_content = request.json

```

Comment:
If we don't use `fiware_node_content` outside of this function, there's no need to make it a global variable.